### PR TITLE
Implement `argv` in `sys` to get command line arguments

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Related: https://github.com/lcompilers/lpython/issues/1358

### CODE REVIEW

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 1]
```diff
@@ -1165,3 +1165,24 @@ LFORTRAN_API void _lpython_close(int64_t fd)
...
+char **argv;
+
+LFORTRAN_API void _lpython_set_argv(int32_t argc_1, char **argv_1) {
+    argv = malloc(argc_1 * sizeof(char *));
+    for (size_t i = 0; i < argc_1; i++) {
+        argv[i] = strdup(argv_1[i]);
+    }
+    argc = argc_1;
+}
+__attribute__((section(".init_array"))) static void *ctr = &_lpython_set_argv;
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/runtime/lfortran_intrinsics.c:1180]: This doesn't seem to work in macOS and Windows.

Comment (certik) [reply src/libasr/runtime/lfortran_intrinsics.c:1180]: Just use regular C, not these `__attribute__` or stuff.

I think we either have to call some `libc` functions to access the arguments, or we need to emit this `_lpython_set_argv` function in the ASR->LLVM/C backends when the program starts.

Comment (certik) [reply src/libasr/runtime/lfortran_intrinsics.c:1180]: Looks like there is no easy way to do that: https://stackoverflow.com/questions/2471553/access-command-line-arguments-without-using-char-argv-in-main#24718544

So I recommend we simply get those arguments in `main` (both in C and in LLVM), and then call our function like `_lcompilers_set_argv`, to set these in our "internal structure", like you did above. That seems like a portable, clean approach that should work everywhere.

**Comment by Thirumalai-Shaktivel** [comment]
The CI failure seems unrelated...

**Comment by Thirumalai-Shaktivel** [comment]
It seems the C backend causes Segfault.

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -393,3 +393,5 @@ RUN(NAME func_static_01 LABELS cpython llvm c wasm)
 RUN(NAME func_static_02 LABELS cpython llvm c wasm)
 RUN(NAME func_dep_03    LABELS cpython llvm c)
 RUN(NAME func_dep_04    LABELS cpython llvm c)
+
+RUN(NAME test_argv_01    LABELS llvm)
```
Comment (certik) [inline integration_tests/CMakeLists.txt:397]: Also enable `cpython` test here.

File: integration_tests/test_argv_01.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from ltypes import i32
+from sys import argv
+
+def test():
+    exe: str = argv()[0]
```
Comment (certik) [inline integration_tests/test_argv_01.py:5]: I think this has to be:
```suggestion
    exe: str = argv[0]
```

File: src/runtime/sys.py [hunk 4]
```diff
@@ -6,3 +6,23 @@ def exit(error_code: i32):
     """
 
     quit(error_code)
+
+def argv() -> list[str]:
+    # TODO: use argv as a global `list[str]` variable
```
Comment (certik) [inline src/runtime/sys.py:11]: I see, it's in the TODO list here.

**Comment by certik** [comment]
This looks good. Now the issue is how to create the argv global variable and initialize it.

One option is to add a global statement like:
```
argv = ...
```

But I think this might not be working yet. In that case I recommend to finish this PR, rename `argv()` to `_lpython_argv()`, merge. Then we will work on the global statements. Then we can do `argv: list[str] = _lpython_argv()` or something like that, and that should work.

How does CPython's `sys.py` does this? Is it implemented in C?

File: integration_tests/test_argv_01.py (hunk 3 above)
Comment (Thirumalai-Shaktivel) [reply integration_tests/test_argv_01.py:5]: Exactly, but global `list` doesn't work yet.

**Comment by Thirumalai-Shaktivel** [comment]
Yup, sys is implementes in C, See: https://stackoverflow.com/a/51580860/15913193

**Comment by certik** [comment]
It's worth studying how they do it. They have this function here https://github.com/python/cpython/blob/91cdd75ba50eea4d0a608324e4a84afab8ae3fbd/Python/sysmodule.c#L2692 and functions above it, that set the global argv list. Are these called when the CPython interpreter/program starts every time?

**Review by certik — APPROVED** [review]
I think this is good to merge as is.

Then we'll improve upon it with subsequent PRs.

Thanks!

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for approving and merging this PR.

**Pull request merged**